### PR TITLE
write debugger docs

### PR DESCRIPTION
**Pull request by devoncarew**
- write debugger docs (fix https://github.com/flutter/devtools/issues/265)
- some tweaks to existing docs for consistency

### CODE REVIEW

File: docs/debugger.md [hunk 1]
```diff
@@ -2,4 +2,69 @@
 title: Using the Debugger
 ---
 
-TODO:
+* toc
+{:toc}
+
+## Getting started
+
+DevTools includes a full source level debugger, including breakpoints, stepping, and
```
Comment (jacob314) [inline docs/debugger.md]: nit: including breakpoints --> supporting breakpoints

File: docs/debugger.md [hunk 2]
```diff
@@ -2,4 +2,69 @@
...
+To adjust the break on exceptions behavior, toggle the `Break on unhandled exceptions`
+and `Break on all exceptions` checkboxes in the upper right of the debugger UI.
+
+Breaking on unhandled exceptions will only pause execution if the breakpoint is considered
+uncaught by the application code. Breaking on all exceptions will cause the debugger to
+pause whether or not the breakpoint was caught by application code.
+
+## Known issues
+
+- when performing a hot restart for a Flutter application, user breakpoints are cleared
```
Comment (jacob314) [inline docs/debugger.md]: Nit: capitalization and punctuation?
- When performing a hot restart for a Flutter application, user breakpoints are cleared.

File: docs/debugger.md (hunk 1 above)
Comment (devoncarew) [reply docs/debugger.md]: done

File: docs/debugger.md (hunk 2 above)
Comment (devoncarew) [reply docs/debugger.md]: done

**Pull request merged**